### DISCUSSION TITLE
Flet testingproject / app creation

### DISCUSSION THREAD

**Discussion by jvalle63** [flet-dev/flet#2003] [Q&A]
### Question

Iam unable to create a project and run a flet app (Android) for testing purposes on windows environment, even folowing flet instructions...

### Code sample

```python
Create a new Flet project:

flet create my-app
cd my-app
###########################################
Run the following command to start Flet development server with your app:

flet run --android
###########################################
usage: flet create [-h] [-v] [--project-name PROJECT_NAME] [--description DESCRIPTION] [--template {minimal}] output_directory

Create a new Flet app from a template.

positional arguments:
  output_directory      project output directory

options:
  -h, --help            show this help message and exit
  -v, --verbose         -v for detailed output and -vv for more detailed
  --project-name PROJECT_NAME
                        project name for the new Flet app
  --description DESCRIPTION
                        the description to use for the new Flet project
  --template {minimal}  template to use for new Flet project
Create a new Flet app from a template.
###########################################
usage: flet run [-h] [-v] [-p PORT] [--name APP_NAME] [-m] [-d] [-r] [-n] [-w] [--ios] [-a ASSETS_DIR] [script]

Run Flet app.

positional arguments:
  script                path to a Python script

options:
  -h, --help            show this help message and exit
  -v, --verbose         -v for detailed output and -vv for more detailed
  -p PORT, --port PORT  custom TCP port to run Flet app on
  --name APP_NAME       app name to distinguish it from other on the same port
  -m, --module          treat the script as a python module path as opposed to a file path
  -d, --directory       watch script directory
  -r, --recursive       watch script directory and all sub-directories recursively
  -n, --hidden          application window is hidden on startup
  -w, --web             open app in a web browser
  --ios                 open app on iOS device
  -a ASSETS_DIR, --assets ASSETS_DIR
                        path to assets directory

positional arguments:
  output_directory      project output directory

options:
  -h, --help            show this help message and exit
  -v, --verbose         -v for detailed output and -vv for more detailed
  --project-name PROJECT_NAME
                        project name for the new Flet app
  --description DESCRIPTION
                        the description to use for the new Flet project
  --template {minimal}  template to use for new Flet project
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
What do you mean by un able please? Can you give further details?

**Reply by jvalle63** [reply]
I need a detailed example, like you have done in other topics, like ft.Textfield, for example, or ft.column

**Reply by jvalle63** [reply]
for example. What does it mean this error?. What I have done wrong?

(env) PS C:\Users\55119\Dropbox\Python\curso-python> cd\my-app
cd\my-app : Não foi possível carregar o módulo 'cd'. Para obter mais 
No linha:1 caractere:1
+ cd\my-app
+ ~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (cd\my-app:String) [], CommandN  
   otFoundException
    + FullyQualifiedErrorId : CouldNotAutoLoadModule
 
(env) PS C:\Users\55119\Dropbox\Python\curso-python> ls


    Diretório: C:\Users\55119\Dropbox\Python\curso-python


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
da----        07/06/2023     14:44                .idea
da----        08/11/2023     18:29                .vs
da----        09/11/2023     15:08                .vscode
da----        17/10/2022     16:52                Aula 24 - Facilitadores     
da----        17/10/2022     16:52                aula 25 - exercicios
da----        17/10/2022     16:52                Aula 37- Maps e outras      
                                                  funçoes
da----        17/10/2022     16:52                Aula15-Exercicios
da----        17/10/2022     16:52                Aula18_while
da----        03/05/2023     11:48                Aula19-listas
da----        17/10/2022     16:52                aula21-split_join_enumerate  
da----        17/10/2022     16:52                aula26-def
da----        17/10/2022     16:52                aula29-dicionarios
da----        17/10/2022     16:52                aula30 - conjuntos-sets      
da----        17/10/2022     16:52                aula31- list comprehension  
da----        17/10/2022     16:52                aula32- dictionary
                                                  comprehension
da----        17/10/2022     16:52                aula33-geradores,iteradores 
                                                  e iteráveis
da----        17/10/2022     16:52                aula35- Zip
da----        17/10/2022     16:52                aula36-itertools
da----        07/10/2023     18:25                aula38-exceções(try
da----        17/10/2022     16:52                aula39-módulos
da----        25/11/2023     14:28                Aula40_manipular_arquivos_ja
                                                  son
da----        17/10/2022     16:52                aula41_path_modulos_pacotes 
da----        17/10/2022     16:52                aula42_decorador
da----        17/10/2022     16:52                aula44_undo_redo
da----        17/10/2022     16:52                aula45_valida_cnpj
da----        03/05/2023     11:48                aula46_poo_classes
da----        17/10/2022     16:52                aula47_context_manager      
da----        17/10/2022     16:52                aula48_doc_strings
da----        28/10/2022     13:38                aula49_exercicio_ipv4       
da----        03/05/2023     11:54                aula51__main__
da----        16/09/2023     14:42                aula52_bibliotea_date_time_m
                                                  oeda_locais
da----        03/05/2023     12:05                aula53_os_sistema
da----        03/05/2023     12:05                aula54_Json
da----        03/05/2023     12:09                Aula55_CSV
da----        03/05/2023     12:03                aula56_random
da----        03/05/2023     12:04                aula57_string_funcoes       
da----        03/05/2023     12:07                aula58_variavel_sistema_emai
                                                  l
da----        03/05/2023     12:03                aula59_zip
da----        03/05/2023     11:55                aula60_sys
da----        06/05/2023     15:58                aula61_web_scratch
da----        01/10/2023     13:51                aula62_html_css
da----        10/11/2023     14:50                env
da----        26/10/2023     18:40                output
da----        09/09/2023     13:51                teste
da----        10/11/2023     13:07                tutorial-env
da----        06/10/2023     14:38                web_scrap
da----        17/06/2022     14:30                __pycache__
-a----        23/04/2023     15:55            224 .env
-a----        16/04/2023     15:04             81 .env-example
-a----        25/11/2023     14:26             36 abc.txt
-a----        10/11/2023     14:38            266 ambiente_virtual.txt
-a----        06/05/2022     13:27            539 aula 27-espessoes Lambda -  
                                                  anonimas.py
-a----        22/02/2022     18:24            532 Aula 4.py
-a----        28/02/2022     13:41            337 Aula 6.py
-a----        24/03/2022     15:05            216 Aula 9.py
-a----        14/02/2022     12:38             25 Aula1.py
-a----        24/03/2022     15:23            325 aula10.py
-a----        24/03/2022     16:51            580 aula11.py
-a----        26/03/2022     15:13            186 aula12.py
-a----        28/03/2022     17:11            856 aula13-validação strings.py  
-a----        28/03/2022     13:35            376 aula14-pass.py
-a----        28/03/2022     18:53            917 aula16.py
-a----        29/03/2022     15:07            686 aula17.py
-a----        18/02/2022     13:02             53 Aula2.py
-a----        04/04/2022     17:19            522 aula20-for_else.py
-a----        11/04/2022     14:38            349 aula22-desempacotamento.py  
-a----        11/04/2022     14:50             77 aula23-invertendo
                                                  variáveis.py
-a----        10/05/2022     12:37            576 aula28-Tuplas.py
-a----        21/02/2022     12:47             28 Aula3.py
-a----        04/06/2022     15:32            633 aula34-exercício
                                                  e-comerce.py
-a----        25/06/2022     12:51           1025 aula43_Problemas_parametros_
                                                  mutaveis.py
-a----        24/02/2022     13:23            198 Aula5.py
-a----        01/11/2022     14:16           1720 aula50_data_classes.py      
-a----        28/02/2022     19:12            983 Aula7.py
-a----        24/03/2022     12:19            464 Aula8.py
-a----        05/05/2023     15:21           3336 config.json
-a----        10/11/2023     13:20             60 curso-python.code-workspace 
-a----        23/10/2023     15:34          27387 hoteis_guarujá - sp_maio de 
                                                  2024_maio de 2024.xlsx       
-a----        23/10/2023     16:07          29140 hoteis_manaus - am_novembro 
                                                  de 2023_novembro de
                                                  2023.xlsx
-a----        23/10/2023     21:42          27893 hoteis_vitoria - es_maio de 
-a----        25/11/2023     13:24             95 jsonP.json
-a----        08/02/2022     14:52            544 main.py
-a----        25/11/2023     15:24             12 select_stat.json
-a----        11/04/2022     13:55            301 Teste 1.py
-a----        29/04/2023     20:34            710 teste.txt


(env) PS C:\Users\55119\Dropbox\Python\curso-python> flet create my-app        
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\55119\Dropbox\Python\curso-python\env\Scripts\flet.exe\__main__.py", line 7, in <module>
  File "C:\Users\55119\Dropbox\Python\curso-python\env\Lib\site-packages\flet\cli\cli.py", line 83, in main
    args.handler(args)
  File "C:\Users\55119\Dropbox\Python\curso-python\env\Lib\site-packages\flet\cli\commands\create.py", line 58, in handle
    cookiecutter(
  File "C:\Users\55119\Dropbox\Python\curso-python\env\Lib\site-packages\cookiecutter\main.py", line 80, in cookiecutter
    base_repo_dir, cleanup_base_repo_dir = determine_repo_dir(
                                           ^^^^^^^^^^^^^^^^^^^
cutter\repository.py", line 107, in determine_repo_dir
    cloned_repo = clone(
                  ^^^^^^
  File "C:\Users\55119\Dropbox\Python\curso-python\env\Lib\site-packages\cookiecutter\vcs.py", line 85, in clone
    raise VCSNotInstalled(msg)
cookiecutter.exceptions.VCSNotInstalled: 'git' is not installed.
(env) PS C:\Users\55119\Dropbox\Python\curso-python> pip install git
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\contourpy-1.2.0-py3.11-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\control-0.7.0-py3.11.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330       
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\cycler-0.12.1-py3.11.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330       
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\fonttools-4.44.3-py3.11.egg is deprecated. pip 24.3 will enforce 
this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330    
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\kiwisolver-1.4.5-py3.11-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\matplotlib-3.8.2-py3.11-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\pillow-10.1.0-py3.11-win-amd64.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package 
installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\55119\dropbox\python\curso-python\env\lib\site-packages\pyparsing-3.1.1-py3.11.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330     
ERROR: Could not find a version that satisfies the requirement git (from versions: none)
ERROR: No matching distribution found for git
(env) PS C:\Users\55119\Dropbox\Python\curso-python>

**Reply by FeodorFitsner** [reply]
Make sure you have Git installed and it's available in PATH.

**Reply by jvalle63** [reply]
ok...
